includes/components/class-geolocation.php

```diff
@@ -112,7 +112,7 @@ public function add_attributes( $attributes ) {
 			[
 				'location'  => [
 					'editable'     => true,
-					'searchable'   => true,
+					'searchable'   => in_array( 'location', (array) get_option( 'hp_listing_search_fields' ) ),
```

Comment:
Please make sure that these and below changes support models other than listings, because Geolocation also support requests, vendors, etc. so instead of hard-coding the listing model name changes should be based on "geolocation_models" option, also note that not all the models can have the "..._search_fields" option so checking this is also needed.